apps/aehttp/test/aehttp_hyperchains_SUITE.erl

```diff
@@ -718,6 +727,79 @@ epoch_with_slow_parent(Config) ->
                  [rpc(?NODE1, aec_chain_hc, epoch_start_height, [N]) || N <- lists:seq(1, EndEpoch)]),
     ok.
 
+%%%=============================================================================
+%%% Pinning
+%%%=============================================================================
+
+get_pin(Config) ->
+
+    %% note: we are actuall in epoch=2, but the pins are for the previous epoch
+    Repl1 = aecore_suite_utils:http_request(aecore_suite_utils:external_address(), get, "hyperchain/pin-tx", []),
+    {ok, 200, #{<<"epoch">> := 1}} = Repl1,
+    %% product, still in same epoch, so pins should be the same
+    {ok, _} = produce_cc_blocks(Config, 2),
+    Repl2 = aecore_suite_utils:http_request(aecore_suite_utils:external_address(), get, "hyperchain/pin-tx", []),
+    {ok, 200, #{<<"epoch">> := 1}} = Repl2,
+    ?assertEqual(Repl1, Repl2),
+    %% move into next epoch, test pin:epoch
+    {ok, _} = produce_cc_blocks(Config, 9),
+    Repl3 = aecore_suite_utils:http_request(aecore_suite_utils:external_address(), get, "hyperchain/pin-tx", []),
+    {ok, 200, #{<<"epoch">> := 2}} = Repl3,
+
+    ok.
+
+post_pin_to_pc(Config) ->
+
+    %% we use local/rpc here.
+    Pin = rpc(?NODE1, aec_pinning_agent, get_pinning_data, []),
+    PinPayloadBin = rpc(?NODE1, aec_pinning_agent, encode_pin_payload, [Pin]),
+
+   DwightPub = pubkey(?DWIGHT), % PC chain account
+   DwightEnc = aeser_api_encoder:encode(account_pubkey, DwightPub),
+   ParentNodeSpec = #{scheme => "http", host => "127.0.0.1", port => aecore_suite_utils:external_api_port(?PARENT_CHAIN_NODE)},
+   {ok, []} = rpc(?PARENT_CHAIN_NODE, aec_tx_pool, peek, [infinity]), % no pending transactions
+   PinTx = aec_pinning_agent:create_pin_tx(ParentNodeSpec, DwightEnc, DwightPub, 1, 30000 * ?DEFAULT_GAS_PRICE, PinPayloadBin), 
+   SignedPinTx = sign_tx(PinTx, privkey(?DWIGHT),?PARENT_CHAIN_NETWORK_ID),
+   {ok, #{<<"tx_hash">> := TxHash}} = aec_pinning_agent:post_pin_tx(SignedPinTx, ParentNodeSpec),
+   {ok, [_]} = rpc(?PARENT_CHAIN_NODE, aec_tx_pool, peek, [infinity]), % one transaction pending now.
+   PinHeight = rpc(?PARENT_CHAIN_NODE, aec_chain, top_height, []),
```

Comment:
No, the height is when the transaction is accepted on chain. Check that it is on chain in a loop and then use that height.